### DISCUSSION TITLE
Assigning Assets and Public URL of assets

### DISCUSSION THREAD

**Discussion by MTerlecky** [pimcore/pimcore#15554] [Q&A / Support]
Hello,

I'm new to PIMcore and am trying to learn how to use it. My employer has installed it for me and I am currently trying to use it as Product Catalog and DAM. It is installed on the web and we are using AWS for image storage. I am using version 10.5.12. I know how to upload images to Assets but I'd like to bulk assign them to Data Objects. Is there a way to do this? Also, when looking at my assets, the grid shows the full path but I'd like to be able to see the public URL on the grid. Right now, the only way I know to view the public URL is to select an image and click the Info button. Is there a way to get the public URL to show up in the grid so I can export it?

Thanks in advance

**Comment by kara-hyyyper** [comment]
I'm not sure about bulk image assignment. I am currently manually dragging each image to each object. 

As for getting the URL for your image, you can create a calculated value that can print the image url.
I followed this guide briefly: https://pimcore.com/docs/platform/Pimcore/Assets/Working_with_Thumbnails/Image_Thumbnails/

After creating my thumbnail,  my formula for creating URLs is as follows:
```PHP
if ($context->getFieldname() == "ImageURL") {
    if ($object->getImage()) {
        return $IP . $object->getImage()->getThumbnail("MyImageURL");
    }
    return "";
}
```
"ImageURL" is the name of the calculated value.
"Image" is the name of the image component.
$IP is a variable I created above with my personal IP.
"MyImageURL" is the name of the Thumbnail I created.
This formula is with in a custom calculator I created, which you will have to figure out how to do if you haven't already.

This prints something like
```
IPAddress/Image Folder/Category/Size/000/image-thumb__000__MyImageURL/ImageName.a0aa0aaa.jpg
```

**Reply by kara-hyyyper** [reply]
And if you make a calculated value that is on all your objects then it also gets exported with all your other variables.
